master/buildbot/www/oauth2.py

```diff
@@ -172,11 +172,17 @@ class GitHubAuth(OAuth2Auth):
     name = "GitHub"
     faIcon = "fa-github"
     authUri = 'https://github.com/login/oauth/authorize'
+    authUriAdditionalParams = {'scope': 'user:email'}
     tokenUri = 'https://github.com/login/oauth/access_token'
     resourceEndpoint = 'https://api.github.com'
 
     def getUserInfoFromOAuthClient(self, c):
         user = self.get(c, '/user')
+        emails = self.get(c, '/user/emails')
+        for email in emails:
+            if email.get('primary', False):
```

Comment:
you debug print here to see why it is not covered. This is not obvious to me though